internal/backends/winit/lib.rs

```diff
@@ -200,3 +228,98 @@ impl i_slint_core::platform::Platform for Backend {
         })
     }
 }
+
+/// Invokes the specified callback with a reference to the [`winit::event_loop::EventLoopWindowTarget`].
+/// Use this to get access to the display connection or create new windows with [`winit::window::WindowBuilder`].
+///
+/// *Note*: This function can only be called from within the Slint main thread.
+pub fn with_event_loop_window_target<R>(
+    callback: impl FnOnce(&winit::event_loop::EventLoopWindowTarget<SlintUserEvent>) -> R,
+) -> R {
+    crate::event_loop::with_window_target(|event_loop_interface| {
+        callback(event_loop_interface.event_loop_target())
+    })
+}
+
+#[doc = concat!("This helper trait can be used to obtain access to the [`winit::window::Window`] for a given [`slint::Window`](https://slint-ui.com/releases/", env!("CARGO_PKG_VERSION"), "/docs/rust/slint/struct.window).")]
+pub trait WinitWindowAccessor {
```

Comment:
Not necessarily in this commit,  but we probably want to seal that trait.

Alternative to a trait would be to make winit_window_rc_for_window public (with another name).
Not sure what's better.

Reply:
I agree about the sealing. Added to my todo next week.

I understand your preference towards a function - a direction that you've guided me towards a few times in the past and one that I kind of agree with. However I think in the context of winit the trait is more idiomatic - it seems the typical pattern used in rust-windowing APIs.

Reply:
I added a sealing commit on top. Does that look correct to you?